### PR TITLE
Dosenbach 2010 ROIs as in paper

### PR DESCRIPTION
**Pull request by salma1601**
I kept the ROIs order as in the original paper. I also have a version of these ROIs ordered by networks, in case this would be preferred ?

### CODE REVIEW

File: nilearn/datasets/atlas.py [hunk 1]
```diff
@@ -616,3 +616,40 @@ def fetch_atlas_aal(version='SPM12', data_dir=None, url=None, resume=True,
               'labels': labels_dict}
 
     return Bunch(**params)
+
+
+def fetch_atlas_dosenbach_2010():
+    """Download and load the Dosenbach et al. brain atlas composed of 160 ROIs
+    grouped in 6 networks.
```
Comment (banilo) [inline nilearn/datasets/atlas.py]: The difference form many other fetchers appears to be gthat the atlas is a set of coordinates and not 3D shapes. That might be mentioned here.

Comment (GaelVaroquaux) [inline nilearn/datasets/atlas.py]: > The difference form many other fetchers appears to be gthat the atlas is a set
> of coordinates and not 3D shapes. That might be mentioned here.

I do wonder if we should call this an atlas? Maybe fetch_coords_dosenbach
rather?

File: nilearn/datasets/atlas.py [hunk 2]
```diff
@@ -616,3 +616,40 @@ def fetch_atlas_aal(version='SPM12', data_dir=None, url=None, resume=True,
               'labels': labels_dict}
 
     return Bunch(**params)
+
+
+def fetch_atlas_dosenbach_2010():
```
Comment (KamalakerDadi) [inline nilearn/datasets/atlas.py]: Just curious to know ```datasets.fetch_abide_pcp(derivatives=['rois_dosenbach160'])``` ?

Is this PR different than we get ?

**Comment by salma1601** [comment]
>I do wonder if we should call this an atlas? Maybe fetch_coords_dosenbach rather?

I changed the function naming. Then  ```fetch_atlas_power_2011``` also should be renamed using coords I think.
I have a problem in the documentation: in the modules reference page only the first sentence is given so it stops at "...Dosenbach et al." . I don't know how to avoid this ?
@KamalakerDadi  ```datasets.fetch_abide_pcp(derivatives=['rois_dosenbach160'])``` gives ABIDE time-series extracted from an atlas of spheres of 4.5 mm radius around these coordinates.

**Comment by GaelVaroquaux** [comment]
> Then fetch_atlas_power_2011 also should be renamed using coords I
> think.

I agree. Do you want to add an issue for that?

**Comment by salma1601** [comment]
>Do you want to add an issue for that?

Yes, I am a passionate nilearn-issue-generator
Done in #954

File: nilearn/datasets/description/dosenbach_2010.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Dosenbach 2010 atlas
+
+
+Notes
+-----
+160 ROIs covering much of the cerebral cortex and cerebellum.
```
Comment (banilo) [inline nilearn/datasets/description/dosenbach_2010.rst]: write out "ROI"

File: nilearn/datasets/description/dosenbach_2010.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+They were obtained from meta-analyses of fMRI activation studies
+and assigned into 6 networks according to a modularity analysis of
```
Comment (banilo) [inline nilearn/datasets/description/dosenbach_2010.rst]: maybe make the type of modularity analysis explicit here, such that one does not need to go back read the paper

Comment (GaelVaroquaux) [inline nilearn/datasets/description/dosenbach_2010.rst]: > maybe make the type of modularity analysis explicit here, such that one does
> not need to go back read the paper

Really? I don't think that we should strive to summarize or explain
methods that we are not implementing.

Comment (banilo) [inline nilearn/datasets/description/dosenbach_2010.rst]: "modularity analysis" does not mean anything -> perhaps add a half sentence that at least is not ambiuous about what will be fetched

File: nilearn/datasets/atlas.py [hunk 5]
```diff
@@ -616,3 +616,39 @@ def fetch_atlas_aal(version='SPM12', data_dir=None, url=None, resume=True,
               'labels': labels_dict}
 
     return Bunch(**params)
+
+
+def fetch_coords_dosenbach_2010():
+    """Load the Dosenbach et al. 160 ROIs. These ROIs cover
+    much of the cerebral cortex and cerebellum and are assigned into 6
+    networks. Coordinates are given in MNI space.
```
Comment (banilo) [inline nilearn/datasets/atlas.py]: maybe move the MNI statement to the returns section

File: nilearn/datasets/atlas.py [hunk 6]
```diff
@@ -616,3 +616,39 @@ def fetch_atlas_aal(version='SPM12', data_dir=None, url=None, resume=True,
               'labels': labels_dict}
 
     return Bunch(**params)
+
+
+def fetch_coords_dosenbach_2010():
+    """Load the Dosenbach et al. 160 ROIs. These ROIs cover
+    much of the cerebral cortex and cerebellum and are assigned into 6
```
Comment (banilo) [inline nilearn/datasets/atlas.py]: grammar: "assigned to"

File: nilearn/datasets/description/dosenbach_2010.rst (hunk 4 above)
Comment (GaelVaroquaux) [inline nilearn/datasets/description/dosenbach_2010.rst]: > > +160 ROIs covering much of the cerebral cortex and cerebellum.
> > +They were obtained from meta-analyses of fMRI activation studies
> > +and assigned into 6 networks according to a modularity analysis of

> "modularity analysis" does not mean anything -> perhaps add a half sentence
> that at least is not ambiuous about what will be fetched

That's the terminology they use in the paper. I would simply leave it
like that, and reference the paper.

**Comment by AlexandreAbraham** [comment]
I think that we only need to fix @banilo's comments. For the example, I was thinking of adding dosenbach to my example with the power atlas in #924. What do you think?

**Comment by salma1601** [comment]
>I was thinking of adding dosenbach to my example with the power atlas in #924.
I was thinking the same thing !
> I think that we only need to fix @banilo's comments
I was waiting for a decision about the naming issue #954. Shall I name it fetch_atlas or fetch_coords or load_coords ??

**Comment by AlexandreAbraham** [comment]
> I was waiting for a decision about the naming issue #954. Shall I name it fetch_atlas or fetch_coords or load_coords ??

The actual policy in nilearn is to name it `load_atlas`. `load` because it is not downloaded and `atlas` because we do not differentiate atlases from coordinates ATM. I suggest that you set this name and we open an issue to discuss atlas vs. coords.

**Comment by KamalakerDadi** [comment]
> I suggest that you set this name

meaning ```fetch_atlas``` ?

**Comment by AlexandreAbraham** [comment]
`load_atlas`

**Comment by AlexandreAbraham** [comment]
OK, Kamalakar is right. I thought we had `load_atlas_power` but it's `fetch`. Let's call it `fetch_atlas`

**Comment by salma1601** [comment]
> I suggest that you set this name and we open an issue to discuss atlas vs. coords.

This is #954.

> Let's call it `fetch_atlas`

Ok, done.

**Comment by AlexandreAbraham** [comment]
Can you fix the conflicts?

**Comment by salma1601** [comment]
Yes I am on it

**Comment by salma1601** [comment]
test_atlas.test_fail_fetch_atlas_harvard_oxford is failing on my labtop.

**Comment by AlexandreAbraham** [comment]
> merge master

To keep history clean, we do not allow merges. Please rebase.

> test_atlas.test_fail_fetch_atlas_harvard_oxford is failing on my labtop.

Trace?

**Comment by salma1601** [comment]
>To keep history clean, we do not allow merges. Please rebase.

Sorry. Is it ok now ?

>Trace?

Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/salma/CODE/Parietal/nilearn/nilearn/datasets/tests/test_atlas.py", line 143, in test_fail_fetch_atlas_harvard_oxford
    assert_equal(len(ho.labels), 5)
AssertionError: 97 != 5
    '97 != 5' = '%s != %s' % (safe_repr(97), safe_repr(5))
    '97 != 5' = self._formatMessage('97 != 5', '97 != 5')
  raise self.failureException('97 != 5')

**Comment by AlexandreAbraham** [comment]
You may have local changes, Travis is happy with this test...

**Pull request merged**